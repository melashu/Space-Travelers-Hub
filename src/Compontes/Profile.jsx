import React from 'react';
import { Table } from 'react-bootstrap';
import { getMissionState } from '../Store/reducers';
import { getRocketState } from '../Store/rocketReducer';
import { useAppSelector } from '../hooks/hooks';
import { getDragonState } from '../Store/dragonreducer';

const Profile = () => {
  const missions = useAppSelector(getMissionState);
  const rockets = useAppSelector(getRocketState);
  const dragons = useAppSelector(getDragonState);
  const filteredMission = missions.filter((mission) => mission.reserved);
  const reserevedMission = filteredMission.map((mission) => (
    <tr key={mission.mission_id}>
      <td>{mission.mission_name}</td>
    </tr>
  ));

  const filteredRocket = rockets.filter((rocket) => rocket.reserved);
  const reserevedRocket = filteredRocket.map((rocket) => (
    <tr key={rocket.id}>
      <td>{rocket.rocket_name}</td>
    </tr>
  ));

  const filteredDragon = dragons.filter((dragon) => dragon.reserved);
  const reserevedDragon = filteredDragon.map((dragon) => (
    <tr key={dragon.id}>
      <td>{dragon.name}</td>
    </tr>
  ));

  return (
    <div className="tables" data-testid="table">
      <div className="rocket">
        <h2>Resireved Rocket</h2>
        <Table hover bordered>
          <tbody>{reserevedRocket}</tbody>
        </Table>
      </div>
      <div className="mission">
        <h2>Resireved Mission</h2>
        <Table hover bordered>
          <tbody>{reserevedMission}</tbody>
        </Table>
      </div>
      <div className="dragon">
        <h2>Resireved Dragon</h2>
        <Table hover bordered>
          <tbody>{reserevedDragon}</tbody>
        </Table>
      </div>
    </div>
  );
};

export default Profile;
