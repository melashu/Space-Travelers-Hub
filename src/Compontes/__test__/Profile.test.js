import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../Store/Store';
import Profile from '../Profile';

const ProfileProvider = () => {
  <Provider store={store}>
    <Profile />
  </Provider>;
};

describe('Should Header render correctly', () => {
  it('Test Profile Componenet', () => {
    const component = renderer.create(<ProfileProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
