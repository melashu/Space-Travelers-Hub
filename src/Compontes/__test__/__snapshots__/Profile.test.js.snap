// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly Test Profile Componenet 1`] = `null`;
